common/beerocks/bcl/include/bcl/network/socket.h

```diff
@@ -43,28 +43,83 @@ typedef unsigned __int64 ssize_t;
 
 class Socket {
 public:
-    explicit Socket(SOCKET s, long readTimeout = 1000);
-    Socket(SOCKET s, const std::string &peer_ip, int port_port, long readTimeout = 1000);
-    Socket(const std::string &uds_path = std::string(), long readTimeout = 1000);
```

Comment:
Since you were able to remove these two constructors without adding a corresponding factory method, I think you should simply remove them in a preparatory commit.

Well actually the last one is still used by the SocketClient and SocketServer classes, but I believe the constructor that takes a peer_ip is unused.